entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -94,10 +113,16 @@ class EntityHandler(
      */
     @GetMapping("/{entityId}", produces = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE])
     fun getByURI(@PathVariable entityId: String): Mono<ResponseEntity<*>> {
-        return entityId.toMono()
+        return extractSubjectOrEmpty().doOnNext {
+            if (!entityService.exists(entityId)) throw ResourceNotFoundException("Entity Not Found")
+            if (!authorizationService.userCanReadEntity(
+                    entityId,
+                    it
+                )
+            ) throw AccessDeniedException("User forbidden read access to entity $entityId")
+        }.then(entityId.toMono())

```

Comment:
Don't feel forced to transform `entityId` to a Mono when you have it ready to be used